freqtrade/optimize/hyperopt.py

```diff
@@ -686,16 +717,21 @@ def start(self) -> None:
                         # evaluations can take different time. Here they are aligned in the
                         # order they will be shown to the user.
                         val['is_best'] = is_best
-
-                        self.print_results(val)
+                        output = self.get_results(val)
+                        if output:
+                            self.progress_bar.write(output)
+                        self.progress_bar.update(1)
 
                         if is_best:
                             self.current_best_loss = val['loss']
                         self.trials.append(val)
                         # Save results after each best epoch and every 100 epochs
                         if is_best or current % 100 == 0:
                             self.save_trials()
+                self.progress_bar.close()
+
         except KeyboardInterrupt:
+            self.progress_bar.close()
```

Comment:
This looks to be a case for `finally:` ...
note that you should check existance of the progressbar (`if self.progress_bar`) - assuming i am able to stop it before it's initialized ...

Reply:
It's better to instantiate tqdm in a `with` statement here -- it will close it automatically when it's needed

Reply:
changing back to progressbar2